pkg/network/usm/ebpf_gotls_helpers.go

```diff
@@ -8,15 +8,131 @@
 package usm
 
 import (
+	"debug/elf"
 	"errors"
 	"fmt"
+	"os"
 	"reflect"
+	"time"
 	"unsafe"
 
+	"github.com/cilium/ebpf"
+	"golang.org/x/sys/unix"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf/uprobes"
 	"github.com/DataDog/datadog-agent/pkg/network/go/bininspect"
 	"github.com/DataDog/datadog-agent/pkg/network/protocols/http/gotls"
+	libtelemetry "github.com/DataDog/datadog-agent/pkg/network/protocols/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+// goTLSBinaryInspector is a BinaryInspector that inspects Go binaries, dealing with the specifics of Go binaries
+// such as the argument passing convention and the lack of uprobes
+type goTLSBinaryInspector struct {
+	structFieldsLookupFunctions map[bininspect.FieldIdentifier]bininspect.StructLookupFunction
+	paramLookupFunctions        map[string]bininspect.ParameterLookupFunction
```

Comment:
1. I think we should move declaration closer to usage, a.k.a move the declaration of `paramLookupFunctions` and `structFieldsLookupFunctions` to the current file
2. I'm not sure it makes a lot of benefit for storing them as struct members rather than accessing them as global definitions within the package

Reply:
I can move them into this file, but I don't think we should use them as globals. If they are structs, the `goTLSBinaryInspector` could easily be adapted by other code that needed to attach to Go binaries.

Reply:
So let's move them to the right file

Reply:
@gjulianm - bump here

Reply:
Fixed